frontend/express/public/javascripts/countly/vue/components/vis.js

```diff
@@ -1594,6 +1594,111 @@
                     </div>'
     }));
 
+    Vue.component("cly-map-picker", countlyVue.components.create({
+        components: {
+            'l-map': Vue2Leaflet.LMap,
+            'l-tile-layer': Vue2Leaflet.LTileLayer,
+            'l-marker': Vue2Leaflet.LMarker,
+            'l-circle': Vue2Leaflet.LCircle,
+        },
+        props: {
+            radius: {
+                type: Object,
+                required: false,
+                validator: function(value) {
+                    return value && (value.unit === 'km' || value.unit === 'mi') && value.value;
+                },
+                default: null
+            }
+        },
+        data: function() {
+            return {
+                markerCoordinates: null,
+                defaultCenterCoordinates: {lat: 48.66194284607008, lng: 8.964843750000002}, // Note: arbitrary coordinates used for map center when user location is not found
+                userCenterCoordinates: null,
+                tileFeed: 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
+                tileAttribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+                MI_TO_KM_RATIO: 1.60934,
+                KM_TO_M_RATIO: 1000,
+                RadiusUnitEnum: {
+                    KM: 'km',
+                    MI: 'mi'
+                }
+            };
+        },
+        computed: {
+            radiusInMeters: function() {
+                if (this.radius && this.radius.unit === this.RadiusUnitEnum.KM) {
+                    return this.radius.value * this.KM_TO_M_RATIO;
+                }
+                if (this.radius && this.radius.unit === this.RadiusUnitEnum.MI) {
+                    return this.radius.value * this.MI_TO_KM_RATIO * this.KM_TO_M_RATIO;
+                }
+                return 0;
+            },
+            centerCoordinates: function() {
+                if (this.markerCoordinates) {
+                    return this.markerCoordinates;
+                }
+                if (this.userCenterCoordinates) {
+                    return this.userCenterCoordinates;
+                }
+                return this.defaultCenterCoordinates;
+            },
+            dynamicZoom: function() {
+                if (this.userCenterCoordinates) {
+                    return 12;
+                }
+                return 6;
+            },
+        },
+        methods: {
+            setMarkerCoordinates: function(coordinates) {
+                this.markerCoordinates = coordinates;
+            },
+            onLocationClick: function(event) {
+                this.setMarkerCoordinates(event.latlng);
+                this.$emit('select', event.latlng);
```

Comment:
We can emit the value, it is ok. How do we get the initial value tho? So I would expect a :value/@input pair here, but may be there is something I'm missing :slightly_smiling_face:

Reply:
You're right.I initially thought that I did not need it but it turns out we are also displaying a specific location that was created using the map.